src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -365,91 +365,69 @@ ParsedQuery Visitor::visit(Parser::UpdateContext* ctx) {
-  } else if (ctx->deleteWhere()) {
-    auto [toDelete, pattern] = visit(ctx->deleteWhere());
-    parsedQuery_.updateClause().toDelete_ = std::move(toDelete);
-    parsedQuery_._rootGraphPattern = std::move(pattern);
+  if (ctx->deleteWhere() || ctx->modify()) {
+    return visitAlternative<ParsedQuery>(ctx->deleteWhere(), ctx->modify());
   } else {
-    visitAlternative<void>(ctx->load(), ctx->drop(), ctx->add(), ctx->move(),
-                           ctx->copy(), ctx->create());
-    AD_FAIL();
+    parsedQuery_._clause = visitAlternative<parsedQuery::UpdateClause>(
+        ctx->load(), ctx->clear(), ctx->drop(), ctx->create(), ctx->add(),
+        ctx->move(), ctx->copy(), ctx->insertData(), ctx->deleteData());
   }
 
   return parsedQuery_;
 }
 
 // ____________________________________________________________________________________
-void Visitor::visit(const Parser::LoadContext* ctx) const {
-  reportNotSupported(ctx, "SPARQL 1.1 Update Load is");
+Load Visitor::visit(Parser::LoadContext* ctx) {
+  return Load{
+      static_cast<bool>(ctx->SILENT()), visit(ctx->iri()),
+      ctx->graphRef() ? visit(ctx->graphRef()) : std::optional<GraphRef>{}};
 }
 
 // ____________________________________________________________________________________
-ParsedQuery Visitor::visit(Parser::ClearContext* ctx) {
-  auto graphRef = visit(ctx->graphRefAll());
-
-  if (holds_alternative<DEFAULT>(graphRef)) {
-    parsedQuery_._clause = parsedQuery::UpdateClause();
-    parsedQuery_.updateClause().toDelete_ = {
-        {Variable("?s"), Variable("?p"), Variable("?o")}};
-    parsedQuery_._rootGraphPattern._graphPatterns.emplace_back(
-        BasicGraphPattern{{{Variable("?s"), "?p", Variable("?o")}}});
-    return parsedQuery_;
-  } else {
-    reportNotSupported(ctx, "Named Graphs are");
-  }
+Clear Visitor::visit(Parser::ClearContext* ctx) {
+  return Clear{static_cast<bool>(ctx->SILENT()), visit(ctx->graphRefAll())};
 }
 
 // ____________________________________________________________________________________
-void Visitor::visit(const Parser::DropContext* ctx) const {
-  reportNotSupported(ctx, "SPARQL 1.1 Update Drop is");
+Drop Visitor::visit(Parser::DropContext* ctx) {
+  return Drop{static_cast<bool>(ctx->SILENT()), visit(ctx->graphRefAll())};
 }
 
 // ____________________________________________________________________________________
-void Visitor::visit(const Parser::CreateContext* ctx) const {
-  reportNotSupported(ctx, "SPARQL 1.1 Update Create is");
+Create Visitor::visit(Parser::CreateContext* ctx) {
+  return Create{static_cast<bool>(ctx->SILENT()), visit(ctx->graphRef())};
 }
 
 // ____________________________________________________________________________________
-void Visitor::visit(const Parser::AddContext* ctx) const {
-  reportNotSupported(ctx, "SPARQL 1.1 Update Add is");
+Add Visitor::visit(Parser::AddContext* ctx) {
+  return Add{static_cast<bool>(ctx->SILENT()), visit(ctx->graphOrDefault()[0]),
```

Comment:
Add a `CORRECTNESS_CHECK`  thatthe size is 2 and use `.at()` instead of `[]` (increase robustness).